### PR TITLE
Improve assertMacroExpansion highlight verification

### PR DESCRIPTION
**Pull request by rauhul**
Adds a new initializer to DiagnosticSpec which takes an optional array of highlights instead of a single optional highlight string. Updates existing initializer to map the string into an array of one.

Updates assertMacroExpansion to check each highlight in the diagnostic spec against the actual produced diagnostic individually. Additionally changes the behavior to drop leading and trailing trivial when performing the highlight comparison to better match the Swift compiler's behavior.

### CODE REVIEW

**Comment by rauhul** [comment]
This is behavior breaking iff:
- you use explicit highlight checking (it seems like most users don't) and
  - your explicit highlights had trailing space to match the misreported highlights from swift-syntax or
  - your diagnostics emit multiple highlights, which now need to be passed as an array instead of string with odd concatenation
  
IMO this should be considered a bug fix because the current highlight checking doesn't actually allow the user to be confident in their highlight assertion.

**Review by ahoppen — COMMENTED** [review]
Looks good to me. Could you add a test case that tests passing multiple `highlights` to `DiagnosticSpec`?

File: Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift [hunk 1]
```diff
@@ -129,7 +129,7 @@ public struct DiagnosticSpec {
   public let severity: DiagnosticSeverity
 
   /// If not `nil`, the text the diagnostic is expected to highlight
```
Comment (ahoppen) [inline Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift:131]:
```suggestion
  /// If not `nil`, the text fragments the diagnostic is expected to highlight
```

Comment (rauhul) [reply Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift]: fixed in `cd73ad23cc9502efcb654c2d3da4c3de95942dfd`

**Comment by rauhul** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by rauhul** [comment]
The linux error seems unrelated to me:
```
error: multiple targets named '_InstructionCounter' in: 'swift-format', 'swift-syntax'; consider using the `moduleAliases` parameter in manifest to provide unique names
```

**Comment by ahoppen** [comment]
Yes, fixed by https://github.com/apple/swift-format/pull/632.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by ahoppen — COMMENTED** [review]
I have two comments about API compatibility. If you could address those and add the change to `Release Notes/510.md`, we can get this merged.

**Comment by rauhul** [comment]
@swift-ci please test

**Comment by rauhul** [comment]
@ahoppen could you take another look when you have some time?

**Comment by rauhul** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
Good to merge 👍🏽

**Comment by rauhul** [comment]
@swift-ci please test

**Pull request merged**